crates/bitwarden-core/src/client/encryption_settings.rs

```
@@ -85,76 +66,58 @@ impl EncryptionSettings {
             // )
         };
 
-        Ok(EncryptionSettings {
-            user_key,
-            private_key,
-            org_keys: HashMap::new(),
-        })
+        #[allow(deprecated)]
+        {
+            let mut ctx = store.context_mut();
+            ctx.set_symmetric_key(SymmetricKeyId::User, user_key)?;
+            if let Some(private_key) = private_key {
+                ctx.set_asymmetric_key(AsymmetricKeyId::UserPrivateKey, private_key)?;
+            }
+        }
+
+        Ok(())
     }
 
     /// Initialize the encryption settings with only a single decrypted key.
     /// This is used only for logging in Secrets Manager with an access token
     #[cfg(feature = "secrets")]
-    pub(crate) fn new_single_key(key: SymmetricCryptoKey) -> Self {
-        EncryptionSettings {
-            user_key: key,
-            private_key: None,
-            org_keys: HashMap::new(),
-        }
+    pub(crate) fn new_single_key(key: SymmetricCryptoKey, store: &KeyStore<KeyIds>) {
+        #[allow(deprecated)]
+        store
+            .context_mut()
+            .set_symmetric_key(SymmetricKeyId::User, key)
+            .expect("Mutable context");
     }
 
     #[cfg(feature = "internal")]
     pub(crate) fn set_org_keys(
-        &mut self,
         org_enc_keys: Vec<(Uuid, AsymmetricEncString)>,
-    ) -> Result<&Self, EncryptionSettingsError> {
-        use bitwarden_crypto::KeyDecryptable;
+        store: &KeyStore<KeyIds>,
+    ) -> Result<(), EncryptionSettingsError> {
+        let mut ctx = store.context_mut();
+
+        if !ctx.has_asymmetric_key(AsymmetricKeyId::UserPrivateKey) {
+            return Err(VaultLockedError.into());
```

**issue:** Technically this can also mean the user doesn't have a private key. I think we're breaking the later PM-11690 return. And we'd want to test this on such an account.

---

Hmm that's weird, previously it was returning `EncryptionSettingsError::MissingPrivateKey` correctly, I have no idea why I'd change it.

I'll change it back to return the older error.

---

I think we need to return this error after, we check if `org_enc_keys` is empty, https://github.com/bitwarden/sdk-internal/pull/8/files/4a6363b988a03d9393a44817993c701896cfae77#diff-673812a5c065e8c33981caaa0b1e1bedc9369bbf318782925769b5a9ae10cf73R109